creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -602,9 +610,52 @@ class SnapbackSM {
             }
           }
 
+          /**
+           * If either secondary has a Sync success rate for user below threshold, add it to `unhealthyReplicas` list
+           */
+          // Fetch secondary SyncRequest outcomes from SecondarySyncHealthTracker
+          const userSecondarySyncHealthOutcomes = await SecondarySyncHealthTracker.getSyncMetricsForUser(nodeUser.wallet)
+
+          // Compute sync success rate per secondary per user
+          let sec1UserSyncSuccesses = 0; let sec1UserSyncFailures = 0; let sec2UserSyncSuccesses = 0; let sec2UserSyncFailures = 0
```

Comment:
coupled with the refactor above this could maybe be generalized a bit so it doesn't effectively parse the redis key which is pretty strong coupling

Reply:
per your slack comment, i think i see what you're saying
basically, instead of returning:
```
{
  "SecondarySyncRequestOutcomes-Daily:::http://cn3_creator-node_1:4002:::0x6f71113010305015351d5b666a1e4317cf6a4a70:::RECURRING:::2021-06-30:::Success": 4,
  "SecondarySyncRequestOutcomes-Daily:::http://cn4_creator-node_1:4003:::0xcf6effeb85571d2895a39610b19051033ddfb523:::MANUAL:::2021-06-30:::Success": 2,
  "SecondarySyncRequestOutcomes-Daily:::http://cn2_creator-node_1:4001:::0x6f71113010305015351d5b666a1e4317cf6a4a70:::RECURRING:::2021-06-30:::Success": 1
}
```

parse it into a more logical structure before returning such as:
```
{
  "wallet": {
    "secondaryUrl": {
      "date1": {
        "SuccessCount": 5,
        "failureCount": 2
      },
      ....etc
    }
  }
}
```

is that correct?

either way, what i did for now is basically moved this logic into the class itself. its not generalized at all but it at least decouples the caller from having to worry about any redis internals. lmk what you think

Reply:
fair point, done - lmk